packages/compass-sidebar/src/components/legacy/sidebar.tsx

```diff
@@ -125,7 +127,7 @@ export function Sidebar({
       if (action === 'copy-connection-string') {
         void copyConnectionString(
           maybeProtectConnectionString(
-            connectionInfo.connectionOptions.connectionString
+            initialConnectionInfo?.connectionOptions?.connectionString ?? ''
```

Comment:
initialConnectionInfo is not optional
```suggestion
            initialConnectionInfo.connectionOptions.connectionString
```